controllers/mattermost/mattermost/mattermost.go

```diff
@@ -142,7 +143,12 @@ func (r *MattermostReconciler) checkMattermostRBAC(mattermost *mmv1beta.Mattermo
 }
 
 func (r *MattermostReconciler) checkMattermostSA(mattermost *mmv1beta.Mattermost, reqLogger logr.Logger) error {
+	if mattermost.Spec.FileStore.External != nil && mattermost.Spec.FileStore.External.UseServiceAccount {
```

Comment:
This is my only nitpick about this PR, if we are using our own ServiceAccount and we are not creating it (which it does at the Mattermost object level), does it makes sense to specify it at the filestore level? Would it make more sense to have it on the Mattermost spec? @mirshahriar feedback on this?

Reply:
I think it makes sense to have it at the external filestore level since it only affects the external file store object.

Reply:
I'm 0/5 on this, but I usually propagate logic "up to bottom" rather than "bottom-up". That is, the service account is for Mattermost itself, not only for the filestore, so, in my mind, the definition of if it should be external or not should be done at the Mattermost level and then propagate it to the underlying requirements, rather than setting it up on a "requirement" and then act accordingly. Unsure if I'm explaining myself properly 😓 

Again, 0/5, this is not a blocker, is just something that I saw during the review, and pretty sure similar things are being done the same in other parts, that's why I wanted extra eyes on this.

Reply:
Yeah, I am also thinking that the service account is a more general configuration to the Mattermost app. As such it should probably be defined in the Mattermost spec section instead of the filestore even if all we use it for is the filestore for now. @fmartingr does this sound like it is lining up with your thoughts as well? I think it is, but just want to check.

Reply:
> Yeah, I am also thinking that the service account is a more general configuration to the Mattermost app. As such it should probably be defined in the Mattermost spec section instead of the filestore even if all we use it for is the filestore for now. @fmartingr does this sound like it is lining up with your thoughts as well? I think it is, but just want to check.

Better explained in less words, yes 🙃

Reply:
@Hunter-Thompson we just chatted offline as a team and have aligned on how to proceed here. We actually came to the consensus that your current implementation is fine as is. Although the service account itself if a lower level resource for the Mattermost app, the bool value for using it for filestore access makes sense to be in the filestore struct like you have it.

Thanks for your patience on this!